tests/Microsoft.Identity.Test.Common/Core/Mocks/MockHelpers.cs

```diff
@@ -303,6 +303,30 @@ public static string CreateIdToken(string uniqueId, string displayableId, string
             return string.Format(CultureInfo.InvariantCulture, "someheader.{0}.somesignature", Base64UrlHelpers.Encode(id));
         }
 
+        public static string CreateIdTokenWithExtraClaim(string uniqueId, string displayableId, string tenantId)
+        {
+            string id = "{\"aud\": \"e854a4a7-6c34-449c-b237-fc7a28093d84\"," +

```

Comment:
For the tests, you can probably use Wilson (https://www.nuget.org/packages/Microsoft.IdentityModel.JsonWebTokens) to create the raw IDToken from claims ...

Reply:
You can use Wilson to create "correct" tokens with signatures etc. Alternatively you can just add some assertions to integration tests, which will ensure you can parse actual ESTS tokens.